src/WingetCreateCLI/Commands/UpdateCommand.cs

```diff
@@ -280,21 +280,42 @@ public async Task<Manifests> UpdateManifestsAutonomously(Manifests manifests)
                 {
                     installerUpdate.IsZipFile = true;
 
-                    // Obtain all possible relative file paths and check if there is a match.
-                    List<string> relativeFilePaths = installerManifest.Installers.SelectMany(i => i.NestedInstallerFiles?.Select(x => x.RelativeFilePath) ?? Enumerable.Empty<string>()).Distinct().ToList();
+                    // Obtain all nested installer files from the previous manifest.
+                    List<NestedInstallerFile> nestedInstallerFiles = installerManifest.Installers.SelectMany(i => i.NestedInstallerFiles ?? Enumerable.Empty<NestedInstallerFile>()).ToList();
                     string extractDirectory = ExtractArchiveAndRetrieveDirectoryPath(packageFile);
+                    installerUpdate.NestedInstallerFiles = new List<NestedInstallerFile>();
 
-                    installerUpdate.RelativeFilePaths = new List<string>();
-
-                    foreach (string relativeFilePath in relativeFilePaths)
+                    foreach (NestedInstallerFile nestedInstallerFile in nestedInstallerFiles)
                     {
-                        if (!File.Exists(Path.Combine(extractDirectory, relativeFilePath)))
+                        // If the previous RelativeFilePath is not an exact match, check if there is a new suitable match in the archive.
+                        if (!File.Exists(Path.Combine(extractDirectory, nestedInstallerFile.RelativeFilePath)))
                         {
-                            Logger.ErrorLocalized(nameof(Resources.NestedInstallerFileNotFound_Error), relativeFilePath);
-                            return null;
-                        }
+                            string relativeFilePath = ObtainMatchingRelativeFilePath(nestedInstallerFile.RelativeFilePath, extractDirectory, packageFile);
+                            if (relativeFilePath == null)
+                            {
+                                return null;
+                            }
+
+                            List<string> relativeFilePathsInsideList = installerUpdate.NestedInstallerFiles.Select(x => x.RelativeFilePath).ToList();

```

Comment:
Nit: This variable name was slightly confusing for me. I think something like `installerUpdateRelativeFilePathList` might be better.